Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -289,39 +289,39 @@ private void UploadDirectoryToGCSBucket(string directory, Dictionary<string, str
         {
             if (string.IsNullOrWhiteSpace(directory) || !Directory.Exists(directory))
             {
-                // do nothing if directory is not valid
+                // Do nothing if the directory is not valid.
                 return;
             }
 
-            // have to append slash to create directory on the cloud
+            // We have to append / at the end to create a directory on the cloud.
             string relativeDirectoryNameWithSlash = relativeDirectoryName + "/";
 
             if (TestObjectExists(Service, Bucket, relativeDirectoryNameWithSlash) && !Force.IsPresent)
             {
-                // if the directory exists and force is not present, throw an error
+                // If the directory exists and force is not present, throw an error.
                 throw new PSArgumentException(
                     $"Storage object '{relativeDirectoryNameWithSlash}' already exists. Use -Force to overwrite.");
             }
 
-            // create a directory on the cloud
+            // Create a directory on the cloud.
             string objContentType = GetContentType(null, metadataDict, UTF8TextMimeType);
             byte[] contentBuffer = Encoding.UTF8.GetBytes(Contents);
             Stream contentStream = new MemoryStream(contentBuffer);
             UploadStreamToGCSObject(contentStream, objContentType, metadataDict, relativeDirectoryNameWithSlash);
 
             foreach (string file in Directory.EnumerateFiles(directory))
             {
-                // upload each individual file
+                // Upload each individual file.
                 UploadStreamToGCSObject(new FileStream(file, FileMode.Open),
                     GetContentType(ContentType, metadataDict, InferContentType(file)),
                     metadataDict,
-                    // get a relative path compared to the root path, also have to replace \ with / so it will be created with correct folder structure
+                    // Get a relative path compared to the root path, also we have to replace \ with / so it will be created with correct folder structure.
                     Path.Combine(relativeDirectoryName, Path.GetFileName(file)).Replace('\\', '/'));
             }
 
             foreach (string subDirectory in Directory.EnumerateDirectories(directory))
             {
-                // recursively upload subfolder, we have to replace \ with /
+                // Recursively upload subfolder and replace \ with /.
                 UploadDirectoryToGCSBucket(subDirectory, metadataDict, Path.Combine(relativeDirectoryName, Path.GetFileName(subDirectory)).Replace('\\', '/'));
```

Comment:
You should definitely break down these subexpressions into other lines.

Generally having more concise code is a good thing. However, if it hurts readability, like in this case, you should err on the side of being more verbose.

Introduce new local variables to make things easier to read. In this case, maybe introduce 2 or 3.